core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -291,8 +292,12 @@ object SparkSubmit {
       } else {
         Nil
       }
+
+    val ivySettings = Option(args.ivySettingsFile).map(SparkSubmitUtils.loadIvySettings).getOrElse(
```

Comment:
@vanzin I thought this would be useful too, but it turns out I haven't missed it at all using this patch for the last few weeks.  It's no big deal to add any "one off" repositories to an xml file if you need that kind of customization.